### DISCUSSION TITLE
How to use mockkStatic on an overriding function

### DISCUSSION THREAD

**Discussion by TSAI-HSIAO-HAN** [mockk/mockk#1141] [Q&A]
I have two overriding function in object `Helper`, `getAccountInfo(): String` and `getAccountInfo(id: String): String`. 

I want to use mockkStatic on getAccountInfo(id: String) by calling `mockkStatic(Helper::getAccountInfo)`, however, Android Studio will show message 'Overload resolution ambiguity. All these functions match'. 

I tried to specify by using reference by `val f: (String) -> String = Helper::getAccountInfo` and passed in `mockkStatic(f)` which did not work. How should I specify the function in this case?

**Comment by JakubHylaScalable** [comment]
It's been a while, but stumbled upon the same problem and ended up figuring it out, so may be useful for another poor soul.

```
        val function: KFunction3<Date, Long, TimeUnit, Boolean?> = Date::isNewerThan
        mockkStatic(function)
```
